notebooks/py/DARTFISH_Pipeline_-_Human_Occipital_Cortex_-_1_FOV.py

```diff
@@ -0,0 +1,255 @@
+spots_df['area'] = np.pi*spots_df['radius']**2
+spots_df.head()
+# EPY: END code
+
+# EPY: START markdown
+# ### QC Plots
+# EPY: END markdown
+
+# EPY: START markdown
+# #### parameter tuning plots
+# EPY: END markdown
+
+# EPY: START code
+min_dist = 0.6
+
+plt.figure(figsize=(10,3))
+
+plt.subplot(131)
+plt.hist(mags, bins=100);
+plt.yscale('log')
+plt.xlabel('barcode magnitude')
+plt.ylabel('number of pixels')
+sns.despine(offset=2)
+plt.vlines(magnitude_threshold, ymin=plt.gca().get_ylim()[0], ymax=plt.gca().get_ylim()[1])
+
+plt.subplot(132)
+spots_df['area'] = np.pi*spots_df.radius**2
+spots_df.area.hist(bins=30);
+plt.xlabel('area')
+plt.ylabel('number of spots')
+sns.despine(offset=2)
+
+plt.subplot(133)
+spots_df.distance.hist(bins=30)
+plt.xlabel('min distance to code');
+plt.vlines(min_dist, ymin=plt.gca().get_ylim()[0], ymax=plt.gca().get_ylim()[1])
+sns.despine(offset=2)
+# EPY: END code
+
+# EPY: START markdown
+# #### Copy number comparisons
+# EPY: END markdown
+
+# EPY: START code
+cnts_starfish = spots_df[spots_df.distance<=min_dist].groupby('target').count()['area']
+cnts_starfish = cnts_starfish.reset_index(level=0)
+cnts_starfish.rename(columns = {'target':'gene', 'area':'cnt_starfish'}, inplace=True)
+
+mrg = pd.merge(bench, cnts_starfish, on='gene', how='left')
+mrg.head(20)
+# EPY: END code
+
+# EPY: START code
+sns.set_context('talk')
+sns.set_style('ticks')
+
+x = mrg.dropna().cnt.values
+y = mrg.dropna().cnt_starfish.values
+r = np.corrcoef(x, y)
+r = r[0,1]
+
+plt.scatter(x, y, 50,zorder=2)
+
+plt.xlabel('Gene copy number Benchmark')
+plt.ylabel('Gene copy number Starfish')
+plt.title('r = {}'.format(r))
+
+sns.despine(offset=2)
+# EPY: END code
+
+# EPY: START markdown
+# #### visualization of rolonies
+# EPY: END markdown
+
+# EPY: START code
+distance_threshold = min_dist
+
+psd = SpotFinder.PixelSpotDetector(
+    codebook=cb,
+    metric='euclidean',
+    distance_threshold=distance_threshold,
+    magnitude_threshold=magnitude_threshold,
+    min_area=area_threshold[0],
+    max_area=area_threshold[1]
+)
+
+spot_intensities, results = psd.find(zero_norm_stack)
+# EPY: END code
+
+# EPY: START code
+area_lookup = lambda x: 0 if x == 0 else results.region_properties[x - 1].area
+vfunc = np.vectorize(area_lookup)
+mask = np.squeeze(vfunc(results.label_image))
+new_image = np.squeeze(results.decoded_image)*(mask > area_threshold[0])*(mask < area_threshold[1])
+
+plt.figure(figsize=(10,10))
+plt.imshow(new_image, cmap = 'nipy_spectral');
+plt.axis('off');
+plt.title('Coded rolonies');
+
+plt.figure(figsize=(10,10))
+plt.imshow(new_image[600:800, 100:300], cmap = 'nipy_spectral');
+plt.axis('off');
+plt.title('Coded rolonies, zoomed in');
+# EPY: END code
+
+# EPY: START markdown
+# ### visualization of matched barcodes
+# EPY: END markdown
+
+# EPY: START code
+# reshape the spot intensity table into a RxC barcode vector
+pbcs = spot_intensities.stack(traces=(Indices.ROUND.value, Indices.CH.value))
```

Comment:
What's `pbcs`?

Reply:
putative barcodes. that's not clear? 😈

Reply:
lol, not to my pea brain ;)

Reply:
we need a good word for putative barcodes. that's going to keep coming up.